pulsar-common/src/test/java/org/apache/pulsar/common/api/PulsarDecoderTest.java

```diff
@@ -56,5 +61,26 @@ public void testChannelRead() throws Exception {
             .handleActiveConsumerChange(any(CommandActiveConsumerChange.class));
     }
 
+    @Test
+    public void testMaxSize() {
```

Comment:
We would need to have tests for when there is a mismatch between client and broker settings. Right now, a client won't attempt to send out a message bigger than 5Mb. 

This should be negotiated in the connect/connected phase.

Also, what would happen if a consumer has a smaller limit and the broker is trying to push a bigger message? Current behavior would be to force close connection and immediately reconnect, in a loop. That won't be appropriate in case of a conf mismatch.